src/component/directive.ts

```diff
@@ -443,7 +443,7 @@ QWeb.addDirective({
     );
     ctx.addLine(`const fiber = w${componentID}.__owl__.currentFiber;`);
     ctx.addLine(
-      `def${defID}.then(function () {if (fiber.isCompleted) {return;} const vnode = fiber.vnode; pvnode.sel = vnode.sel; ${createHook}});`
+      `def${defID}.then(function () { if (fiber.isCompleted) { if (!w${componentID}.__owl__.isDestroyed) { w${componentID}.destroy(); } return; } const vnode = fiber.vnode; pvnode.sel = vnode.sel; ${createHook}});`
```

Comment:
you can simply call the `destroy` method, which will co the check anyway. we might as well try to keep the directive small